FBSnapshotTestCase/FBSnapshotTestCase.m

```diff
@@ -11,17 +11,35 @@
 #import <FBSnapshotTestCase/FBSnapshotTestCase.h>
 #import <FBSnapshotTestCase/FBSnapshotTestController.h>
 
+@interface FBSnapshotTestCase ()
+  @property (nonatomic, strong, nonnull) NSString *folderName;
+@end
+
 @implementation FBSnapshotTestCase
 {
   FBSnapshotTestController *_snapshotController;
 }
 
 #pragma mark - Overrides
 
+- (instancetype)initWithInvocation:(NSInvocation *)invocation
+{
+  if (self = [super initWithInvocation:invocation]) {
+
+    self.folderName = [self getFolderName];
+  }
+  return self;
+}
+
+- (NSString * _Nonnull)getFolderName
```

Comment:
`_Nonnull` can be omitted here, it's only needed in the interface or the property definition